### PR TITLE
Add support for sorting dependency exclusions

### PR DESCRIPTION
**Pull request by Stephan202**
The implementation of this new `sortExclusions` setting closely mimics the existing `sortDependencies` setting.

I tried to stay as close as possible to the existing code format and architecture.

### CODE REVIEW

File: sorter/src/main/java/sortpom/SortPomImpl.java [hunk 1]
```diff
@@ -77,6 +77,12 @@ private void warnAboutDeprecatedArguments(SortPomLogger log, PluginParameters pl
         if (pluginParameters.sortDependencies.isDeprecatedValueFalse()) {
             log.warn("[DEPRECATED] The 'false' value in sortDependencies is not used anymore, please use empty value '' or omit sortDependencies instead. In the next major version 'true' or 'false' will cause an error!");
         }
+        if (pluginParameters.sortExclusions.isDeprecatedValueTrue()) {
+            log.warn("[DEPRECATED] The 'true' value in sortExclusions is not used anymore, please use value 'groupId,artifactId' instead. In the next major version 'true' or 'false' will cause an error!");
+        }
+        if (pluginParameters.sortExclusions.isDeprecatedValueFalse()) {
+            log.warn("[DEPRECATED] The 'false' value in sortExclusions is not used anymore, please use empty value '' or omit sortDependencies instead. In the next major version 'true' or 'false' will cause an error!");
+        }
```
Comment (Stephan202) [inline sorter/src/main/java/sortpom/SortPomImpl.java]: I wasn't sure about this. One alternative is e.g. to directly throw an exception for this case in `AbstractParentMojo#execute`, but then the `true`/`false` handing is spread over multiple places.

**Comment by Stephan202** [comment]
I see coverage dropped somewhat. I don't think I missed any `sortExclusions` tests for which there is a `sortDependencies` counterpart, but let me know if I missed anything, of if there's a specific test you'd like to see added.

**Comment by Ekryd** [comment]
Hi
I will look at this as soon as I am able. Could you please provide a brief description of the need for this functionality.

**Comment by Ekryd** [comment]
Great addition and well within the scope of the plugin.

**Comment by Ekryd** [comment]
Hi,
Merged into new branch. Will look into the code (and make slight changes). 
I think that the name sortExclusions could be a bit misleading, wouldn't sortDependencyExclusions be better?

File: sorter/src/main/java/sortpom/SortPomImpl.java (hunk 1 above)
Comment (Ekryd) [reply sorter/src/main/java/sortpom/SortPomImpl.java]: Hmmm, it is not right, but the alternative is worse. I guess that we could create a subclass structure for PluginParameters and that would just complicate the code. We let it be for now.

Comment (Ekryd) [reply sorter/src/main/java/sortpom/SortPomImpl.java]: Now I know, true or false in sortDependencyExclusions will fail the build instead

**Comment by Stephan202** [comment]
> Wouldn't sortDependencyExclusions be better?

That also covers plugin dependency exclusions, so I agree that's better :+1: 

>  Merged into new branch. Will look into the code (and make slight changes).

Tnx :) (I did tick the box that should have given you write access to `PicnicSupermarket:feature/sort-exclusions`, but this works too :) )

**Comment by Ekryd** [comment]
I will change the name and commit the code in a few minutes.

I do have task for you. Please add a new test SortDependencyExclusionsTest; it should _at least_ contain the following test methods:

- Test that exception is thrown when true is set
- Test that exception is thrown when false is set
- Test that a pom file is sorted with groupId first
- Test that a pom file is sorted with artifactId first 

Have a look at the file SortDependenciesTest. You decide if you want two test files under sort and verify; or if one file under sort is enough.

**Comment by Ekryd** [comment]
Btw. Great job with pull request! Was it hard to extend with the new functionality?

**Comment by Stephan202** [comment]
> I do have task for you.

No problem; I'll add this to my TODO list; can't guarantee I'll get around to it today/this week, but will make an effort.

> Was it hard to extend with the new functionality?

Extendting the code wasn't too hard; it want mostly grepping and pattern-matching. One thing that did take me a while to figure out was how to interpret the (very lengthy) integration test failure messages, but once I realized that I could just `vimdiff` the relevant files under the `target` directory that wasn't too much of a problem either. Lastly, I did have the feeling that this was more of an exercise in copy-paste-adjust than it should have been, but I did not dive deep enough into the code to give concrete recommendations on how to resolve that.

**Comment by Ekryd** [comment]
Hi, I just merged some code from master into this branch. Please make a pull to get latest changes.

**Comment by Stephan202** [comment]
Tnx for the heads up @Ekryd!

NB: I did not forget; still have this page open in several tabs :wink:. Very busy right now, so it might be a while before I circle back to this.

**Comment by Ekryd** [comment]
Hi!
I'm fixing the tests for this feature now. Seems wasteful not to include functioning code :-)
Will probably to a release during the weekend.

**Comment by Ekryd** [comment]
Released 2.15.0!

**Comment by Stephan202** [comment]
@Ekryd thanks a lot for pulling this one across the finish line! Can't say I'm any less busy than I was back in February, so this is much appreciated :bow:.

**Pull request merged**